plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -226,15 +226,24 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         )
     }
 
-    private fun sendValidationResultTelemetry(validationResult: ValidationResult) {
-        if (!validationResult.valid) {
+    private fun sendValidationResultTelemetry(validationResult: ValidationResult, projectOnLoad: Boolean = false) {
+        // Old telemetry event to be fired only when users click on transform
+        if (!validationResult.valid && !projectOnLoad) {
             CodetransformTelemetry.isDoubleClickedToTriggerInvalidProject(
                 codeTransformPreValidationError = validationResult.invalidTelemetryReason.category ?: CodeTransformPreValidationError.Unknown,
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 result = Result.Failed,
                 reason = validationResult.invalidTelemetryReason.additonalInfo
             )
         }
+        // New projectDetails metric should always be fired whether the project was valid or invalid
+        CodetransformTelemetry.projectDetails(
+            codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+            result = if (!validationResult.valid) Result.Failed else Result.Unknown,
```

Comment:
Why `Result.Unknown`? Also maybe use the `?` ... `:` syntax here rather than embedded if-else

Reply:
I used Unknown to help represent undefined like we would in Typescript/Javascript. For Jetbrains there is no ternary operator in Kotlin and we must use the if/else syntax.

Reply:
Yep just realized no ternary operator exists whoops. My point on the first part was that do we not have `Result.Succeeded`?

Reply:
This isn't a success metric. Its a NOP for reporting. We should keep success for an action that is successful. 

Thinking ahead we would use Success for a runtime validation and this first check is just reporting.